VERDICT: REQUEST_CHANGES

Fix the formatting failure before merge; the implementation otherwise matches the PR 19 runtime-control surface.

messages.go:264: lint is failing on gofmt for the SDKControlRequestBody field block. Run gofmt on messages.go so the new MCPServerName/Enabled/Servers fields align with the surrounding struct formatting. CI currently reports: messages.go:264:1: File is not properly formatted (gofmt).

Checked against sdk.d.ts L1009-L1025 and L2105-L2141: McpSetServersResult tags, method signatures, subtype strings, serverName vs server_name split, enabled:false handling, nil/empty servers -> {}, and JSON-tag response parsing all look correct. The required unit coverage is present, including enabled:false, nil and empty server maps, empty result collections, and control-error propagation.

Overall: Requesting changes only for the gofmt/lint failure.

--- a/messages.go
+++ b/messages.go
@@ -259,12 +259,10 @@ type SDKControlRequestBody struct {
 	Model                  string                              `json:"model,omitempty"`                  // For set_model
 	MaxThinkingTokens      *int                                `json:"max_thinking_tokens,omitempty"`    // For set_max_thinking_tokens
 	UserMessageID          string                              `json:"user_message_id,omitempty"`        // For rewind_files
-	// mcp_message uses snake_case server_name; other MCP control subtypes use
-	// camelCase serverName, matching the CLI wire protocol.
-	ServerName             string                              `json:"server_name,omitempty"`            // For mcp_message
-	MCPServerName          string                              `json:"serverName,omitempty"`             // For MCP control subtypes that use camelCase
-	Enabled                *bool                               `json:"enabled,omitempty"`                // For mcp_toggle
-	Servers                *map[string]MCPServerConfig         `json:"servers,omitempty"`                // For mcp_set_servers
+	ServerName             string                              `json:"server_name,omitempty"`            // For mcp_message (snake_case)
+	MCPServerName          string                              `json:"serverName,omitempty"`             // For mcp_reconnect/mcp_toggle/mcp_set_servers (camelCase)
+	Enabled                *bool                               `json:"enabled,omitempty"`                // For mcp_toggle (pointer so explicit false serializes)
+	Servers                *map[string]MCPServerConfig         `json:"servers,omitempty"`                // For mcp_set_servers (pointer so nil/empty round-trips as {})
 	Message                map[string]interface{}              `json:"message,omitempty"`                // For mcp_message (JSONRPC)
 }
 

